lib/money/money.rb

```diff
@@ -207,11 +207,20 @@ def to_d
   end
 
   def to_s(style = nil)
-    case style
+    units = case style
     when :legacy_dollars
-      sprintf("%.2f", value)
+      2
     when :amount, nil
-      sprintf("%.#{currency.minor_units}f", value)
+      currency.minor_units
+    else
+      return nil
```

Comment:
any reason we would not want to raise here? We expect style to be one of `:legacy_dollars`, `:amount`, or nil.